### PR TITLE
45816967 ADNI Vision Transformer

### PR DESCRIPTION
**Pull request by KennyBoyo**
Hi,

This is 45816967's pull request to add a vision transformer to topic-recognition for the classification of ADNI dataset achieving a test accuracy of 64.43%.

The pull request contains:

- modules.py: The code for the models used for the vision transformer
- dataset.py: The code used to load the data from a folder
- train.py: The code used to run the training loop for the vision transformer
- predict.py: Code used to run predictions from a chosen model
- driver.py: Script to quickly train and predict using the model
- README.md: Project Documentation

Thanks,
Jun Khai Chooi
45816967

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Good work
- Contents explained well
- Plenty of hyperparameter tuning


Commit messages: Exellent

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- Did patient-level split
- Test acc is 64.43%


General comments:
- Solves the problem well although performance is poor
- Good work

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Merging not possible until the gitignore file is removed, please fix. Does not affect grade only merge.

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
